### PR TITLE
Add command for migrating database

### LINKED ISSUE #4948: CLI command for migrating database
There should be a command for migrating the database so that you can run `gitea admin create-user` before starting the server.

We are planning to use it with kubernetes init containers, so that we can start gitea with a predefined user and setup OAuth before accepting requests.

### PR DESCRIPTION
**Pull request by rstefan1**
Fixes #4948

### CODE REVIEW

File: cmd/migrate.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Copyright 2014 The Gogs Authors. All rights reserved.
```
Comment (adelowo) [inline cmd/migrate.go]: This should be `2018 The Gitea Authors` I think since it is a new file

File: Gopkg.lock [hunk 2]
```diff
@@ -1001,6 +1019,8 @@
     "gopkg.in/macaron.v1",
     "gopkg.in/testfixtures.v2",
     "strk.kbt.io/projects/go/libravatar",
+  
   ]
   solver-name = "gps-cdcl"
   solver-version = 1
+
```
Comment (jonasfranz) [inline Gopkg.lock]: Please revert these changed lines

**Review by jonasfranz — DISMISSED** [review]

File: cmd/migrate.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+// license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/models/migrations"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+	"github.com/urfave/cli"
```
Comment (lafriks) [inline cmd/migrate.go:13]: Add new line between gitea imports and external one

**Review by lafriks — CHANGES_REQUESTED** [review]

**Review by lafriks — APPROVED** [review]

File: Gopkg.lock (hunk 2 above)
Comment (rstefan1) [reply Gopkg.lock]: @JonasFranzDEV  solved

**Review by axifive — APPROVED** [review]

**Comment by lunny** [comment]
@JonasFranzDEV need your approval.

**Comment by techknowlogick** [comment]
@JonasFranzDEV need you approval

**Review by appleboy — APPROVED** [review]

**Pull request merged**